packages-native/popup-menu/src/PopupMenu.tsx

```diff
@@ -0,0 +1,90 @@
+import { ComponentType, createElement, ReactElement, useCallback, useEffect, useMemo, useRef } from "react";
+import { PopupMenuProps } from "../typings/PopupMenuProps";
+import { defaultPopupMenuStyles, PopupMenuStyle } from "./ui/Styles";
+import { executeAction } from "@widgets-resources/piw-utils";
+import {
+    BackHandler,
+    Platform,
+    TouchableNativeFeedback,
+    TouchableNativeFeedbackProps,
+    TouchableOpacity,
+    TouchableOpacityProps,
+    View
+} from "react-native";
+import { ActionValue } from "mendix";
+import Menu, { MenuDivider, MenuItem } from "react-native-material-menu";
+import deepmerge from "deepmerge";
+
+export function PopupMenu(props: PopupMenuProps<PopupMenuStyle>): ReactElement {
+    const styles = useMemo(
+        () => deepmerge.all<PopupMenuStyle>([defaultPopupMenuStyles, ...props.style]),
+        [props.style]
+    );
+
+    const menuRef = useRef<any>(null);
+    const showMenu = useCallback(() => {
+        menuRef.current?.show();
+    }, [menuRef]);
+    const handlePress = useCallback(
+        (action?: ActionValue) => {
+            menuRef.current?.hide();
+            executeAction(action);
+        },
+        [menuRef]
+    );
+
+    useEffect(() => {
+        const subscription = BackHandler.addEventListener("hardwareBackPress", () => {
+            menuRef.current?.hide();
+            return undefined;
+        });
+        return () => subscription.remove();
+    }, [menuRef]);
+
+    const Touchable: ComponentType<TouchableNativeFeedbackProps | TouchableOpacityProps> =
+        Platform.OS === "android" ? TouchableNativeFeedback : TouchableOpacity;
+
+    let menuOptions: ReactElement[];
+    if (props.renderMode === "basic") {
+        menuOptions = props.itemsBasic.map((item, index) =>
+            item.itemType === "divider" ? (
+                <MenuDivider key={index} color={styles.dividerColor} />
+            ) : (
+                <MenuItem
+                    key={index}
+                    onPress={() => handlePress(item.action)}
+                    underlayColor={styles.menuItem?.underlayColor}
+                    textStyle={styles.menuItem?.textStyle}
+                >
+                    {item.caption}
+                </MenuItem>
+            )
+        );
+    } else {
+        const touchableProps =
+            Platform.OS === "android"
+                ? {
+                      background: TouchableNativeFeedback.SelectableBackground(),
```

Comment:
We will probably need to allow other props to be manipulated aswell, i would say for now lets not do it until we get the Building blocks, since it has been a tradition where we change the styles and the interfaces billion times

Reply:
Hmm, not sure if we should allow this to be configurable since it expects a complex object